### PR TITLE
[Merged by Bors] - Add workaround for PoET Round 3

### LINKED ISSUE #4967: node configuration: fetch poet proofs from all poets for round 3
## Description
Due to a hardware issue no node was able to submit to PoET 111 for round 3. This PoET has been restored with the state of PoET 110. By default nodes only query poets for proof where they were successfully able to submit their challenge to.

For the upcoming end of PoET round 3 (2023-09-17 20:00 UTC) this behaviour shall be changed such that all nodes that submitted to 110 also query their proofs from PoET 111 instead of only from the ones they were able to submit to.

## Implementation hints
See https://github.com/spacemeshos/go-spacemesh/pull/4752 for reference

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Closes #4967

## Changes
- For PoET round 3 / publish epoch 4 PoET 111 will be added to the local state to fetch proofs from if the node registered to PoET 110 for that round.

## Test Plan
- Added tests to ensure workaround works for the correct epoch / round

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
+
+### Upgrade information
+
+### Highlights
+
+### Features
+
+### Improvements
+
+* [#4969](https://github.com/spacemeshos/go-spacemesh/pull/4969) Make nodes fetch from PoET 111 for round 3 if they were able to register to PoET 110.
```
Comment (poszu) [inline CHANGELOG.md]: I'm not sure if it falls into "improvements" category 😅

**Review by poszu — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (fasmat) [reply CHANGELOG.md]: I'll move it to Features, I'm not sure where else to to put it 🤔

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**